Create training collateral

nrichers opened the pull request:
## Internal Notes for Reviewers

This PR demonstrates a standard for our first training modules, with an eye towards solving an important piece of the training puzzle: how to enable hands-on training for our users. 

Features:

- **Interactive** — it's instructional content combined with the live product.
- **Easy to use** — feels like a slide deck but you're learning about the product while using it.
- **Flexible** — From surfacing the live product, to running Jupyter notebook code right inside our training content, to having our full docs framework available, to fine-grained CSS control, there's a lot we can build with this. 
- **Single sourced** — task steps are shared with our current docs via some single sourcing.
- **Easy to author** — with the exception of the CSS framework extension, we're simply recombining stuff we already have. 

I want you to test this training module and tell me what you think. It's not perfect — the intent here is to improve on the static Google Slides we've been using and to pave the way for future refinements. 

## How can I try this?

Since we trialled an earlier version of this with GBC, the training modules are already live: 

- [Training for Model Developers](https://docs.validmind.ai/training/training-for-model-developers.html)
- [Training for Model Validators](https://docs.validmind.ai/training/training-for-model-validators.html)
- [Training for Administrators](https://docs.validmind.ai/training/training-for-administrators.html)

If you test locally: `gh pr checkout 200`

- The Tachyons CSS framework should be included, but if not:  https://github.com/nareal/tachyons
- Also in this PR but not currently used, an extension for flyouts: https://github.com/schochastics/quarto-nutshell

## Bugs, issues, known problems 

These need to be fixed before merging: 

- [x] JupyterHub
    - [x] Enable iframe embeds for JupyterHub (OR include rendered intro notebook in docs site as a reference)
    - [x] ~OAuth cookie error~ → TRACKED VIA https://app.shortcut.com/validmind/story/4946/fix-oauth-error-in-jupyterhub-for-training-modules
    - [x] Investigate TOC links issue
- [x] Add Navigation hints to slides 
- [x] Add role prerequisites
- [x] Turn off chalkboard to prevent Backspace weirdness
- [x] Single sourcing
    - [x] Fix single-sourced content step numbering
    - [x] Remove “Edit or delete comments” from single source to simplify collaboration slide instructions
- [x] ~Split users & permissions pages (recent UI change):~ → TRACKED VIA https://app.shortcut.com/validmind/story/4437/documentation-design-settings-permissions
    - [x] ~Check if add roles has changed?~
    - [x] ~Check if manage permissions has changed~
- [x] ~Content blocks (recent UI change):~  → TRACKED VIA https://app.shortcut.com/validmind/story/4971
    - [x] ~Update content block docs to match latest UI changes~
    - [x] ~Add edit content blocks section and include as tab in developer training module~
- [x] Update the training home page with direct learning track links and add administrator track
- [x] ~Add a slide with more detail around the workflow prerequisites, e.g. explore the current workflow and then tell developers to look in the administrator training for the details on how to set it up.~  → TRACKED VIA https://app.shortcut.com/validmind/story/4972
- [x] ~"One specific comment for Slide 12/28 on validator track: there needs to be more guidance around the term review."~ → TRACKED VIA https://app.shortcut.com/validmind/story/4973

<!--
PR instructions for release notes:

1. Pick at least one label:

- `internal` (skip Step 2, no release notes required)
- `highlight`
- `enhancement`
- `bug`
- `deprecation`
- `documentation`

2. In the next section, describe the changes so that an external user can understand them. Keep it simple and link to the docs with [Learn more ...](<relative-link>), if available.
-->

## External Release Notes

We're introducing the first training modules that will be part of our training program for model developers, validators, and administrators. Our training modules are interactive, combining instructional content with our live product, and are easy to use. Try it: [Welcome to ValidMind Academy](../../training/training-overview.qmd)

validbeck commented on site/training/training-for-model-validators.qmd:
```diff
@@ -0,0 +1,359 @@
+## {background-iframe="https://app.prod.validmind.ai" background-interactive="true" data-preload="yes"}
+
+::: {.absolute bottom=15 .w-100 .f4 .tc .pl4 .bg-light-pink .ba .b--black-20 .bw1 .br3 .near-black .shadow-5}
+**Try it:** Review the documentation for the **[Quickstart] Customer Churn Model**. | Click [{{< fa chevron-right >}}]() to continue.
```
```suggestion
**TRY IT:** Review the documentation for the **[Quickstart] Customer Churn Model**. {{< fa ellipsis >}} Click [{{< fa chevron-right >}}]() to continue.
```

validbeck commented on site/training/training-for-model-validators.qmd:
```diff
@@ -0,0 +1,359 @@
+
+:::: {.columns}
+::: {.column width="30%"  .pr4}
+Locate the test results in the documentation, review the data, and identify issues with the model. 
```
**GLOBAL COMMENT:** Same idea here about the sizing and colour, and moving the `Try it [...]` to it's own line beneath the two grid columns.

nrichers replied:
> **GLOBAL COMMENT:** Same idea here about the sizing and colour, and moving the `Try it [...]` to it's own line beneath the two grid columns.

Will transfer your othher suggestion globally. The `Try it ...` text is in the left column only when there is otherwise not enough room underneath the two columns.

validbeck reviewed (changes requested):
Nice! I like the functionality of this a lot. Just left some comments about styling tweaks about colours and sizing/placement (and one weird functionality quirk about the back arrow), but otherwise very solid!

nrichers commented on site/training/training-for-model-validators.qmd:
```diff
@@ -0,0 +1,359 @@
+
+:::: {.columns}
+::: {.column width="30%"  .pr4}
+Verify model documentation is complete and complies with regulatory standards by cross-checking against your guidelines. 
```
Pink is generally for links, so we probably can't use that but your formatting is much better overall, thank you. I will transfer these suggestions into the source today.

There are a few formatting-related updates I have in my notes that need to be addressed: 

- Footer text should have solid background when the product UI is shown and better padding where instructions scoll.
- Need to replace HTML buttons with Tachyons CSS (requires `custom.scss`)
- Need to fix offset for steps vs heading for single sourced content (requires `custom.scss`)
- Need to tweak tabset look & feel to match

nrichers commented on site/training/training-for-model-validators.qmd:
```diff
@@ -0,0 +1,359 @@
+## {background-iframe="https://app.prod.validmind.ai" background-interactive="true" data-preload="yes"}
+
+::: {.absolute bottom=15 .w-100 .f4 .tc .pl4 .bg-light-pink .ba .b--black-20 .bw1 .br3 .near-black .shadow-5}
+**Try it:** Review the documentation for the **[Quickstart] Customer Churn Model**. | Click [{{< fa chevron-right >}}]() to continue.
```
> I notice we are doing a little hack here where we use an empty link to change the colour of the chevron to pink... surely there is a better way? 😅

The links will never work once you start interacting with the product UI as the input focus gets shifted to the background layer. If you know how to work around this, I'd love to hear about it!

> LOBAL COMMENT: I think the | divider is a bit inelegant too

It was a late addition to deal with the focus shifting. I don't think the raised elipsis necessarily looks any cleaner, but the next version of this might move that out from the text into another part of the callout with some extra CSS.

nrichers replied:
I ended up just removing the text altogether. Between the nonfunctional link and the need to go taller to make this look OK, it's just too much of a pain right now. There's a lingering navigation issue with the different layers you interact with the product UI layuer regardless, so might be good to take a step back and revisit what our preferred navigation should be.

nrichers commented on site/training/training-for-model-validators.qmd:
```diff
@@ -0,0 +1,359 @@
+::: {.column width="30%"  .pr4}
+Have a question about the model? Collaborate with your developer right in the model documentation.
+
+<br>Try it yourself on the next page. {{< fa hand-point-right >}}
```
> Not sure if there's any way to get the tabbed panels for the collaborate info into one line as well by adjusting only that text size.

I edited the existing headings in the single-sourced content to shorten them as much as I could, but one of the outstanding to-dos is to style the tabs better. They're kinda ugly generally and a smaller font would be helpful.

nrichers commented on site/training/training-overview.qmd:
```diff
@@ -0,0 +1,140 @@
+
+1. Get your access credentials from ValidMind. 
+
+   If you haven't reached out to us yet, email <info@validmind.ai> to enquire about access to the training environment.
```
It's a Google group: https://groups.google.com/a/validmind.ai/g/info — I don't have access to it.

nrichers commented on site/training/training-overview.qmd:
```diff
@@ -0,0 +1,140 @@
+
+Please note that we make updates to the models and datasets available in the training environment from time to time to provide you with our latest features. 
+
+### Sample notebooks 
```
It's a remnant of 'this is the old sandbox stuff from a previous engagement, repurposed as a training landing page. It globally needs more work — would be a good discussion for us to have when we outline all the training program requirements and wish list items.

nrichers commented on site/training/training-overview.qmd:
```diff
@@ -0,0 +1,140 @@
+::: {.callout-important}
+## For testing and evaluation purposes only
+
+Do not upload proprietary information — the training environment environment is not to be used for production. If you have questions about testing in the training environment, please [provide feedback or ask for help](#provide-feedback-or-get-help).
```
It's because you'd be exposing information that should not be uploaded into a training environment — InfoSec would have a field day. 

One thing missing still is a repeatable, consistent training environment that we can periodically wipe or update. It's coming, along with a dedicated training organization on ValidMind, it's just not quite ready yet. Having that will also allows us to use deep links for many of the `TRY IT` pages that currently take you to the model inventory, e.g. where we tell you to do something with some section of the documentation you already located in an earlier step.

noosheenv commented:
My feedback is more in the perspective of a first-time user, so these were just the spots where I got lost in the training.
1. I really like the first few slides, because they're super engaging and aren't too full of information. However, once we get to this slide, I feel a bit lost. After making the context switch however, future slides with this same format weren't so jarring.
![image](https://github.com/validmind/documentation/assets/143854391/fdaf79d8-a7da-405c-8f19-1cf2fd851102)

2. In the commenting section, I was good for following the first instruction, but when I get to this step: "Review your comments on this page", I'm not in the same page as before. So I think switching "this page" to "the home page" or similar might be the move here.
![image](https://github.com/validmind/documentation/assets/143854391/ec78a743-5058-4808-94ad-d8611f6546ce)

3. For the Model findings, there are some issues with the UI that affect my happiness so I'm not sure how much we can do here. When following these instructions step-by-step, I still found myself quite lost, since there are two ways to add (your first) model finding. 
- BTW, based on some restrictions, adding a finding from the home page might be bugged. You have to select the documentation section, but it's only showing None since we're not on any model in the home page. So when you try to submit the finding, you get an error. 
- Also, I can't recreate this bug because the option to add a finding from the homepage disappears after adding a finding from somewhere else! And since we can't delete findings, the option is gone forever!
![image](https://github.com/validmind/documentation/assets/143854391/516c0b26-ae7c-4fbb-ac19-d66a4276d242)

- So I would prefer that the instructions direct the user to create a model finding through the documentation window as their only option

4. I was following the Link findings instructions step-by-step, and found some inconsistencies with the UI behaviour and the instructions. 
- In step 2, it says to click 2. Validation, but that option is not clickable. Since the next instruction says to select 2.1.1 Assumptions and that works, it could be safe to skip that first step.
- In my first read-through, I couldn't tell that we need to get into a page to link a finding. I think the word "jump" makes it sound like we're jumping to a different part of the same page.
- Step 3 asks to hover over a section to link a finding, but the option is already there and nothing happens when you hover.
![image](https://github.com/validmind/documentation/assets/143854391/2bc31b83-e360-47b0-9838-f05a0459630c)

- Step 4 and 5 reference some pages that open after linking a finding, but these pages did not open for me, the UI might have changed?
![image](https://github.com/validmind/documentation/assets/143854391/adb3b69f-f544-43c3-8f2a-ec26cb416770)

- This graphic is outdated but I don't think it's a big deal
![image](https://github.com/validmind/documentation/assets/143854391/e7fb8f5a-c0e7-4a91-bd0a-32d954a9645b)

BTW I like the borders and drop shadows in the reference images. It's professional but it makes them stand out!

nrichers commented:
> 1. I really like the first few slides, because they're super engaging and aren't too full of information. However, once we get to this slide, I feel a bit lost. After making the context switch however, future slides with this same format weren't so jarring.

Great feedback, Noosheen! I think the trick will be to keep the text to a minimum as you make the switch to this format. There's simply not enough room for overlays on the product UI to contain this, but a lot of our steps could be edited to clean them up. (Along the lines of: I'd have written a shorter letter if I had more time.)

>  2\.... when I get to this step: "Review your comments on this page", I'm not in the same page as before.

Great catch! Totally true, fixed the wording. 

> 3\. For the Model findings, there are some issues with the UI that affect my happiness so I'm not sure how much we can do here. When following these instructions step-by-step, I still found myself quite lost, since there are two ways to add (your first) model finding.

Good point. One thing that might help is that we will have more artifacts like realistic findings and more content pre-populated in the UI when we sort out more of our training environment. That will hopefully help bridge the gap a bit, even if the UI experience today is not perfect yet. 

> * BTW, based on some restrictions, adding a finding from the home page might be bugged.

This worked for me when I tested it just now. You said you can't re-create but maybe demo the steps you followed at our next catchup?

> So I would prefer that the instructions direct the user to create a model finding through the documentation window as their only option

We actually pull these instructions straight out of the docs, so we would have to do some rework there. Definitely feasible but more of a future wish list item.

> In step 2, it says to click 2. Validation, but that option is not clickable.

Great catch! It looks like with a recent revamp of how documentation is shown, the docs no longer work and since we single-source the content, the training is wrong, too. Fixed, in three different places no less! 

> In my first read-through, I couldn't tell that we need to get into a page to link a finding. I think the word "jump" makes it sound like we're jumping to a different part of the same page.

These docs are actually for an older version of the UI and the entire paragraph was wrong. Fixed and simplied by referring to expanding the section (was wrong in three different places).

nrichers commented:
> Step 3 asks to hover over a section to link a finding, but the option is already there

Globally fixed this by removing references to hovering, in four places total. 

> Step 4 and 5 reference some pages that open after linking a finding, but these pages did not open for me, the UI might have changed?

These steps worked for me when I tried it just now — you might have to demo this to make sure my addled brain understands. Possibly, you are missing a permission but we'd have to check.

> This graphic is outdated but I don't think it's a big deal

Yup! There are several outdated images, e.g. the workflow one is also very stale. Since we're just surfacing docs content in these steps, we'll need to fix that separately.

nrichers commented on site/training/training-overview.qmd:
```diff
@@ -0,0 +1,140 @@
+
+::: {.g-col-6}
+
+### Model risk management & governance
```
As mentioned elsewhere, one thing missing still is a repeatable, consistent training environment that we can periodically wipe or update. It's coming, along with a dedicated training organization on ValidMind, it's just not quite ready yet. 

Andres and I have discussed the requirements several times and it will include standard registered models, model documentation and validation reports already populated or in progress, and meaningful findings, reports, etc. That will also allow much better linking from some of the `TRY IT` pages. Stay tuned!

nrichers commented on site/training/training-overview.qmd:
```diff
@@ -0,0 +1,140 @@
+ 
+### Join our Slack community
+
+Have feedback or questions? We sponsor a Slack community where you can provide feedback or ask questions: [Join Our Community Slack](https://docs.validmind.ai/guide/join-community.html).
```
Imma stubbornly resist making text a callout just to emphasize it. 😛 If anything, we should emphasize where to get help, which is the Slack channel, but for now I will leave this as-is. Less is more!

nrichers commented on site/training/training-for-model-validators.qmd:
```diff
@@ -0,0 +1,359 @@
+
+<br>
+
+## In this training module
```
Yeah, I was debating the preferred approach here. I think you're right, I will remove the links. 

One thing that would be helpful on this page is some navigational tips for reveal.js. E.g. if you open the hamburger menu, you can jump to any section easily.

nrichers commented on site/training/training-for-model-validators.qmd:
```diff
@@ -0,0 +1,359 @@
+
+From here, you have access to: 
+
+::: {.columns .f3}
```
I like these changes! They feel more like ValidMind to me while being different enough to clearly stand out. Will change globally.

nrichers replied:
Also reworked the modal as suggested — agreed, looks much better.

nrichers commented on site/training/training-for-model-validators.qmd:
```diff
@@ -0,0 +1,359 @@
+
+## {background-iframe="https://app.prod.validmind.ai/model-inventory" background-interactive="true" data-preload="yes"}
+
+::: {.fr .f4 .mv7 .nr5 .pa4 .bg-light-pink .ba .b--black-20 .bw1 .br3 .near-black.}
```
Agreed, implemented this new color scheme globally. Looks great, thank you!

nrichers replied:
I'll leave the width as is for now, as there are issues with the reactive layout in lower resolutions (cf. my demo today what happens).

MichaelIngvarRoenning reviewed (changes requested):
see newer comment.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/validmind/documentation?pullRequest=200) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/validmind/documentation?pullRequest=200) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: nrichers<br/>:x: MichaelIngvarRoenning<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/validmind/documentation?pullRequest=200) it.</sub>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/validmind/documentation?pullRequest=200) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/validmind/documentation?pullRequest=200) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: nrichers<br/>:x: MichaelIngvarRoenning<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/validmind/documentation?pullRequest=200) it.</sub>

MichaelIngvarRoenning reviewed (commented):
A few points:

- It took me quite some time to understand that each screen was interactive that I could click and scroll in the UI and on overall slides that had more information below the starting screen of the slide. Make it very clear that the user needs to interactively move around the slide.

- For both the developer and validator track we are using a ValidMind instance with a lot of models registered in the inventory. There are a lot of weird names, duplicates etc. and I suspect it's hard for a user to locate what the description is telling them to locate. We need a clean up of the model inventory or we need a "clean" instance.

- I think maybe we need to include an admin slide in Session 1 because workflows need to be set up and configured for a model before we can continue the documentation process. I feel like that is a more natural flow of things. Feel free to disagree of course.

- One specific comment for Slide 12/28 on validator track: there needs to be more guidance around the term review.  When I go and click on the validation report for the model stated in description I see the standard report but how should they know what to actually do in the task, is it review model documentation (which is huge) or is it review the validation report template and just click through?

nrichers commented on site/training/training-for-model-developers.qmd:
```diff
@@ -0,0 +1,393 @@
+### SESSION 1
+
+- Initialize the developer framework
+- Start the model development process
```
@MichaelIngvarRoenning we can update this, but the current text essentially repeats what's in the introductory notebook for devs. If we change here, we should change there as well.

nrichers commented on site/training/training-for-model-developers.qmd:
```diff
@@ -0,0 +1,393 @@
+::: {.column width="50%"}
+### SESSION 1
+
+- Initialize the developer framework
```
I agree but we should make this change across all the content for consistency. This text currently lives in all our customer-facing  notebooks.

nrichers commented on site/training/training-for-model-developers.qmd:
```diff
@@ -0,0 +1,393 @@
+
+:::: {.fr .f4 .mv6 .nr5 .pa4 .bg-near-white .ba .b--dark-pink .bw1 .br3 .near-black.}
+
+From **Model Inventory**:
```
This is a great suggestion — it helps with explaining the steps you need to take while doing the UI walkthrough — added to slide 7 where there is more space.

nrichers commented:
> * Make it very clear that the user needs to interactively move around the slide.

Good point, the slides were not explicit enough about interactivity. We now include a callout on the "In this module" page that tells people training is interactive and provides navigation hints. 

> We need a clean up of the model inventory or we need a "clean" instance.

That's the plan! We now have a training org that includes a smaller set of models, see the screenshot. Eventually, we will also have:

- Sample model documentation in progress that you can explore without having to run the developer framework first
- Sample validation report artefacts you can use, such as sample findings, links to evidence, etc. 
- Sample reports for the future MRM governance training

Together, these things will provide a more realistic training environment that mimicks how you would use ValidMind in production.


> I think maybe we need to include an admin slide in Session 1 because workflows need to be set up and configured for a model before we can continue the documentation process. I feel like that is a more natural flow of things.

For GBC for tomorrow, I will leave things as is but perhaps it would make sense to add a slide with more detail around the prerequisites, e.g. explore the current workflow and then tell developers to look in the administrator training for the details on how to set it up. 

(For context, AFAIK devs can't configure the workflow themselves and workflow setup is a pretty big chunk of the training for administrators.)

> One specific comment for Slide 12/28 on validator track: there needs to be more guidance around the term review. 

That's a good point — we're short on guidance not just in the training but in the docs generally. I'll follow up with @juanmleng to see if I can pick his brain on what's entailed in the validator review.

validbeck commented:
# Global comments
- I like `_revealjs_navigation.qmd`... is there a way to somehow make this includes something users can expand again in case they forget the shortcut without needing to go all the way back to this slide?
- I see you already changed some of the "Jupyter Hubs" to `JupyterHub` – thank you!
-  The bottom modules look MUCH better. I like the newer stacked look, very clean.

## Training for model validators
>  Is this lil guy supposed to be transparent? 
<img width="446" alt="Screenshot 2024-06-13 at 9 34 57 AM" src="https://github.com/validmind/documentation/assets/164545837/90db3662-e23a-4b2d-82be-84a41024b546">

## Training for model developers
>In the "ValidMind Introduction for Model Developers" notebook, some of the tables are a bit wonky with the horizontal scroll. Any way we can use some of the white space to the left opened up by no sidenav? Is there a "full width" version we can toggle in Quarto? It looks like [we can only set it at the site level](https://quarto.org/docs/output-formats/page-layout.html#page-layout), hmm...
<img width="843" alt="Screenshot 2024-06-13 at 9 39 03 AM" src="https://github.com/validmind/documentation/assets/164545837/b9d0c62f-8eae-41ff-8b03-187ce6d4bbc7">

## Training for administrators
> Nothing to fix here from a preliminary click-thru. 🎉

nrichers commented:
> * I like `_revealjs_navigation.qmd`... is there a way to somehow make this includes something users can expand again in case they forget the shortcut without needing to go all the way back to this slide?

AFAIK, no, but I've been toying with the idea of adding `? for keyboard shortcuts` to all slides, just haven't done it yet. 

> Is this lil guy supposed to be transparent?

Good catch! Looks like some late-night copy pasta. Should be fixed in https://github.com/validmind/documentation/pull/200/commits/519e0e6d57a94b01d9df04c473ae7c50fee19e22

> In the "ValidMind Introduction for Model Developers" notebook, some of the tables are a bit wonky with the horizontal scroll. 

The current setup is a workaround and I agree it has some issues with how the content is rendered. I hope to overlay the training right over JupyterHub when it's all said and done.

validbeck commented:
> AFAIK, no, but I've been toying with the idea of adding ? for keyboard shortcuts to all slides, just haven't done it yet.

This [plugin](https://revealjs.com/plugins/) looks promising: https://github.com/McShelby/reveal-helpbutton

![image](https://github.com/validmind/documentation/assets/164545837/4bae0267-555c-400f-a386-3079d10733b9)

> Should be fixed in https://github.com/validmind/documentation/commit/519e0e6d57a94b01d9df04c473ae7c50fee19e22

🎉 Can confirm!

nrichers commented:
> This plugin looks promising: https://github.com/McShelby/reveal-helpbutton

I'm going to park this in a new story, but a couple of comments:

- We might not need a plugin at all, we can just add a help link with our own stuff 
- Not all reveal.js plugins work in Quarto. In fact many don't — my one insight after two days of testing CSS framework options. What's listed on quarto.org are the safe ones, plus there are a few other ones floating around. 

That said, I'm all for testing and tweaking, and improving the training MVP iteratively. If that's something you want to tinker with, just claim the story!
